integration-cli/docker_api_netmode_test.go

```diff
@@ -0,0 +1,191 @@
+	if out, _, err = runCommandWithOutput(runCmd); err != nil {
+		c.Fatalf(out, err)
+	}
+	if !checkPS(out) {
+		c.Fatalf("Expected PS output, got %s", out)
+	}
+
+	runCmd = exec.Command(dockerBinary, "run", "--net=host", "busybox", "ps")
+	if out, _, err = runCommandWithOutput(runCmd); err != nil {
+		c.Fatalf(out, err)
+	}
+	if !checkPS(out) {
+		c.Fatalf("Expected PS output, got %s", out)
+	}
+
+	runCmd = exec.Command(dockerBinary, "run", "-h=name", "--net=bridge", "busybox", "ps")
+	if out, _, err = runCommandWithOutput(runCmd); err != nil {
+		c.Fatalf(out, err)
+	}
+	if !checkPS(out) {
+		c.Fatalf("Expected PS output, got %s", out)
+	}
+
+	runCmd = exec.Command(dockerBinary, "run", "-h=name", "--net=none", "busybox", "ps")
+	if out, _, err = runCommandWithOutput(runCmd); err != nil {
+		c.Fatalf(out, err)
+	}
+	if !checkPS(out) {
+		c.Fatalf("Expected PS output, got %s", out)
+	}
+
+	runCmd = exec.Command(dockerBinary, "run", "-h=name", "--net=host", "busybox", "ps")
+	if out, _, err = runCommandWithOutput(runCmd); err == nil {
+		c.Fatalf(out, err)
+	}
+	if !strings.Contains(out, runconfig.ErrConflictNetworkHostname.Error()) {
+		c.Fatalf("Expected error containing '%s' got %s", runconfig.ErrConflictNetworkHostname, out)
+	}
+
+	runCmd = exec.Command(dockerBinary, "run", "-h=name", "--net=container:other", "busybox", "ps")
+	if out, _, err = runCommandWithOutput(runCmd); err == nil {
+		c.Fatalf(out, err)
+	}
+	if !strings.Contains(out, runconfig.ErrConflictNetworkHostname.Error()) {
+		c.Fatalf("Expected error containing '%s' got %s", runconfig.ErrConflictNetworkHostname, out)
+	}
+
+	runCmd = exec.Command(dockerBinary, "run", "--net=container", "busybox", "ps")
+	if out, _, err = runCommandWithOutput(runCmd); err == nil {
+		c.Fatalf(out, err)
+	}
+	if !strings.Contains(out, "--net: invalid net mode: invalid container format container:<name|id>") {
+		c.Fatalf("Expected error containing '%s' got %s", "--net: invalid net mode: invalid container format container:<name|id>", out)
+	}
+
+	runCmd = exec.Command(dockerBinary, "run", "--net=weird", "busybox", "ps")
+	if out, _, err = runCommandWithOutput(runCmd); err == nil {
+		c.Fatalf(out, err)
+	}
+	if !strings.Contains(out, "invalid --net: weird") {
+		c.Fatalf("Expected error containing '%s' got %s", "invalid --net: weird", out)
+	}
+}
+
+func (s *DockerSuite) TestConflictContainerNetworkAndLinks(c *check.C) {
+	var (
+		out    string
+		err    error
+		runCmd *exec.Cmd
+	)
+
+	runCmd = exec.Command(dockerBinary, "run", "--net=container:other", "--link=zip:zap", "busybox", "ps")
+	if out, _, err = runCommandWithOutput(runCmd); err == nil {
+		c.Fatalf(out, err)
+	}
+	if !strings.Contains(out, runconfig.ErrConflictContainerNetworkAndLinks.Error()) {
+		c.Fatalf("Expected error containing '%s' got %s", runconfig.ErrConflictContainerNetworkAndLinks.Error(), out)
+	}
+
+	runCmd = exec.Command(dockerBinary, "run", "--net=host", "--link=zip:zap", "busybox", "ps")
```

Comment:
That's unix specific, no reason it can't run from Windows client to Linux daemon. OK if I just remove TestLinksNetworkHostContainer?